core/dbt/include/global_project/macros/materializations/models/incremental/strategies.sql

```diff
@@ -0,0 +1,79 @@
+{% macro get_incremental_append_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_append_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+
+{% macro default__get_incremental_append_sql(arg_dict) %}
+
+  {% do return(get_insert_into_sql(arg_dict["target_relation"], arg_dict["temp_relation"], arg_dict["dest_columns"])) %}
+
+{% endmacro %}
+
+
+{# snowflake #}
+{% macro get_incremental_delete_insert_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_delete_insert_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+{% macro default__get_incremental_delete_insert_sql(arg_dict) %}
+
+  {% do return(get_delete_insert_merge_sql(arg_dict["target_relation"], arg_dict["temp_relation"], arg_dict["unique_key"], arg_dict["dest_columns"])) %}
+
+{% endmacro %}
+
+
+{# snowflake, bigquery, spark #}
+{% macro get_incremental_merge_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_merge_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+{% macro default__get_incremental_merge_sql(arg_dict) %}
+
+  {% do return(get_merge_sql(arg_dict["target_relation"], arg_dict["temp_relation"], arg_dict["unique_key"], arg_dict["dest_columns"])) %}
+
+{% endmacro %}
+
+
+{% macro get_incremental_insert_overwrite_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_insert_overwrite_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+{% macro default__get_incremental_insert_overwrite_sql(arg_dict) %}
+
+  {% do return(get_insert_overwrite_merge_sql(arg_dict["target_relation"], arg_dict["temp_relation"], arg_dict["dest_columns"], arg_dict["predicates"])) %}
+
+{% endmacro %}
+
+
+{% macro get_incremental_default_sql(arg_dict) %}
+
+  {{ return(adapter.dispatch('get_incremental_default_sql', 'dbt')(arg_dict)) }}
+
+{% endmacro %}
+
+{% macro default__get_incremental_default_sql(arg_dict) %}
+
+  {% do return(get_incremental_append_sql(arg_dict)) %}
+
+{% endmacro %}
```

Comment:
This represents an implicit breaking change for maintainers of existing adapter plugins who use the default incremental materialization:
- Previously, the (implicit) default was `delete+insert`
- Now, the explicit default is `append`

I think it's a good change! It's just one we'll want to document very clearly

(cc @dataders)

Reply:
Opened a docs issue: https://github.com/dbt-labs/docs.getdbt.com/issues/1761

Reply:
The changes in `incremental.sql` made me wonder the same thing about default behavior.